src/proto/h1/io.rs

```diff
@@ -233,13 +237,18 @@ where
             if let WriteStrategy::Flatten = self.write_buf.strategy {
                 return self.poll_flush_flattened(cx);
             }
+
             loop {
-                // TODO(eliza): this basically ignores all of `WriteBuf`...put
-                // back vectored IO and `poll_write_buf` when the appropriate Tokio
-                // changes land...
-                let n = ready!(Pin::new(&mut self.io)
-                    // .poll_write_buf(cx, &mut self.write_buf.auto()))?;
-                    .poll_write(cx, self.write_buf.auto().bytes()))?;
+                debug_assert!(
+                    self.io.is_write_vectored(),
+                    "using vectored writes on an IO that does not provide fast vectored \
+                     write support, this is a bug"
+                );
+                let n = {
+                    let mut iovs = [IoSlice::new(&[]); MAX_BUF_LIST_BUFFERS];
```

Comment:
I'd probably set this to 64 instead, since if the transfer is using chunked encoding, each buf is actually a few bufs, with the chunked framing wrapped around each.

Reply:
done in 6914263